presto-main/src/main/java/com/facebook/presto/sql/planner/PlanNodeCanonicalInfo.java

```diff
@@ -85,12 +100,62 @@ public static List<CanonicalPlanWithInfo> getCanonicalInfo(
                 }
                 PlanNode statsEquivalentPlanNode = node.getStatsEquivalentPlanNode().get();
                 Optional<String> hash = planCanonicalInfoProvider.hash(session, statsEquivalentPlanNode, strategy, true);
-                Optional<List<PlanStatistics>> inputTableStatistics = planCanonicalInfoProvider.getInputTableStatistics(session, statsEquivalentPlanNode, strategy, true);
-                if (hash.isPresent() && inputTableStatistics.isPresent()) {
```

Comment:
inputTableStatistics can be empty if we skip the input statistics check during query registration